bionic/persistence.py

```diff
@@ -259,7 +254,9 @@ def _save_or_reregister_result(self, result):
 
         if not local_entry.exactly_matches_query:
```

Comment:
If `local_entry` exactly matches the query, we should set it to `self._stored_local_entry`. Otherwise the stored local entry would still be flushed.

Same for `cloud_entry`.